### PR TITLE
cmake: fix package download with ninja

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

**Comment by akohlmey** [comment]
The `CheckIfNinjaSupportsFortran.cmake` file seems to be missing and it would be more consistent with the variables that it sets, if the file was called `CheckGeneratorSupport.cmake` and then it could have multiple flags that are all set to false by default and enabled according to build tool in particular.
We should then also output in the summary at the end, what "generator" is used and what specific build command (i.e. the value of `${CMAKE_MAKE_PROGRAM}`)

**Comment by junghans** [comment]
@akohlmey thanks, file and summary added.

**Comment by akohlmey** [comment]
@junghans great. Thanks a lot. I've tested it in a variety of configurations and had to make a few tweaks to address some remaining issues. This looks to me like it is ready to be included.

File: cmake/Modules/Packages/KIM.cmake [hunk 1]
```diff
@@ -34,26 +34,22 @@ if(PKG_KIM)
...
       URL_MD5 6ee829a1bbba5f8b9874c88c4c4ebff8
       BINARY_DIR build
-      CMAKE_ARGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+      CMAKE_ARGS ${CMAKE_REQUEST_PIC}
+                 -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
                  -DCMAKE_Fortran_COMPILER=${CMAKE_Fortran_COMPILER}
                  -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
                  -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                 -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
```
Comment (junghans) [inline cmake/Modules/Packages/KIM.cmake]: Is the `-G` option automatically inherited?

File: cmake/Modules/Packages/MSCG.cmake [hunk 2]
```diff
@@ -19,8 +16,15 @@ if(PKG_MSCG)
...
-      CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR> ${CMAKE_REQUEST_PIC} ${EXTRA_MSCG_OPTS}
-      BUILD_COMMAND make mscg INSTALL_COMMAND ""
+      CMAKE_ARGS ${CMAKE_REQUEST_PIC} ${EXTRA_MSCG_OPTS}
+                 -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                 -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                 -DCMAKE_Fortran_COMPILER=${CMAKE_Fortran_COMPILER}
+                 -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+                 -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                 -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+      BUILD_COMMAND ${CMAKE_MAKE_PROGRAM} libmscg.a INSTALL_COMMAND ""
```
Comment (junghans) [inline cmake/Modules/Packages/MSCG.cmake]: I think we should use `${CMAKE_COMMAND} --build . --target libmscg.a` as it is more generic.

File: cmake/Modules/Packages/USER-SCAFACOS.cmake [hunk 3]
```diff
@@ -4,6 +4,7 @@ if(PKG_USER-SCAFACOS)
 
   find_package(GSL REQUIRED)
   find_package(PkgConfig QUIET)
+  find_package(MPI REQUIRED)
```
Comment (junghans) [inline cmake/Modules/Packages/USER-SCAFACOS.cmake]: Why?

Comment (junghans) [inline cmake/Modules/Packages/USER-SCAFACOS.cmake]: Indentation

**Comment by junghans** [comment]
With cmake-3.10 we can recommend the user to run `cmake --build .` instead of `make` or `ninja` directly.

File: cmake/Modules/Packages/KIM.cmake (hunk 1 above)
Comment (akohlmey) [reply cmake/Modules/Packages/KIM.cmake]: Yes, otherwise, I would not have to pass CMAKE_MAKE_PROGRAM along.

Comment (junghans) [reply cmake/Modules/Packages/KIM.cmake]: I found it, too:
```
   ``CMAKE_GENERATOR <gen>``
      Override the CMake generator used for the configure step. Without this
      option, the same generator as the main build will be used. This option is
      ignored if a custom configure command has been specified with the
      ``CONFIGURE_COMMAND`` option.
```

File: cmake/Modules/Packages/MSCG.cmake (hunk 2 above)
Comment (junghans) [reply cmake/Modules/Packages/MSCG.cmake]: Ok, @akohlmey that also fixed the build error the CI found.

**Pull request merged**